Added groups (#942)
codatio/codat-docs: docs/using-the-api/managing-companies.md

### Create a company

> To create a new company, use the [Create company](/platform-api#/operations/create-company) endpoint and provide a name you want to attribute to it in the request body. Parameter `name` is a required parameter to execute this request. You can also provide a `description` to store additional information about the company.

Reviewer: ```suggestion
To create a new company, use the [Create company](/platform-api#/operations/create-company) endpoint and provide a name for the company in the request body.  The `name` parameter is mandatory to execute this request. You can also provide a `description` to store additional information about the company.
```

## Deleting companies

You can delete a company and its data using the [Delete company](/platform-api#/operations/create-company) endpoint.

<Tabs>

<TabItem value="nodejs" label="TypeScript">

```javascript
const companyDeleteResponse = await platformClient.companies.delete({
    companyId: companyCreatedRes.company.id,
  }); 
```
</TabItem>

<TabItem value="python" label="Python">

```python
company_delete_response = platform_client.companies.delete(
  operations.DeleteCompanyRequest(
    company_id=company_created_res.company.id,
  )
)
```
</TabItem>

<TabItem value="csharp" label="C#">

```c#
var companyDeleteResponse = await platformClient.Companies.DeleteAsync(new(){
    CompanyId = companyCreatedRes.Company.Id,
});
```
</TabItem>

<TabItem value="go" label="Go">

```go
ctx := context.Background()
companyDeleteResponse, err := s.Companies.Delete(ctx, operations.DeleteCompanyRequest{
    CompanyID: companyCreatedRes.Company.ID,
})
```
</TabItem>

</Tabs>

## Assign companies to a group

You can group several companies to simplify the management of these companies and their use cases.

> 

Reviewer: ```suggestion
```

## Deleting companies

You can delete a company and its data using the [Delete company](/platform-api#/operations/create-company) endpoint.

<Tabs>

<TabItem value="nodejs" label="TypeScript">

```javascript
const companyDeleteResponse = await platformClient.companies.delete({
    companyId: companyCreatedRes.company.id,
  }); 
```
</TabItem>

<TabItem value="python" label="Python">

```python
company_delete_response = platform_client.companies.delete(
  operations.DeleteCompanyRequest(
    company_id=company_created_res.company.id,
  )
)
```
</TabItem>

<TabItem value="csharp" label="C#">

```c#
var companyDeleteResponse = await platformClient.Companies.DeleteAsync(new(){
    CompanyId = companyCreatedRes.Company.Id,
});
```
</TabItem>

<TabItem value="go" label="Go">

```go
ctx := context.Background()
companyDeleteResponse, err := s.Companies.Delete(ctx, operations.DeleteCompanyRequest{
    CompanyID: companyCreatedRes.Company.ID,
})
```
</TabItem>

</Tabs>

## Assign companies to a group

> Codat supports grouping companies together to simplify the management of companies and their use cases.

Reviewer: ```suggestion
You can group several companies to simplify the management of these companies and their use cases.
```

> If you are grouping companies together for the first time you will need to start by creating a group.

Reviewer: ```suggestion
If you are grouping companies for the first time, you need to create a group first.
```

> A group can be created using the [Create group](/platform-api#/operations/create-group) endpoint.

Reviewer: ```suggestion
```

<Tabs>

<TabItem value="nodejs" label="TypeScript">

```javascript
Create group code snippet
```
</TabItem>

<TabItem value="python" label="Python">

```python
Create group code snippet
```
</TabItem>

<TabItem value="csharp" label="C#">

```c#
Create group code snippet
```
</TabItem>

<TabItem value="go" label="Go">

```go
Create group code snippet
```
</TabItem>

</Tabs>

:::caution Create new groups with care

> Codat does not support the updating or deletion of groups so make sure to double check the name is correct.

Reviewer: ```suggestion
It's not possible to update or delete existing groups, so double-check that the group's name is correct when creating it.
```

:::

> Once you have created a group a company can be assigned to it in twos ways: either at point of company creation or after it is created.

Reviewer: ```suggestion
Once you have created a group, you can assign a company to it in two ways: either at the point of company creation or after it has been created.
```

> Use the [Create company](/platform-api#/operations/create-company) endpoint and pass the `groupId` you want to register the company against.

Reviewer: ```suggestion
For example, use the [Create company](/platform-api#/operations/create-company) endpoint and pass the `groupId` that you want to add the company to.
```

    name='Platypus Properties',
    groups=[shared.GroupRef(id=groupId)]
    )

company_created_res = platform_client.companies.create(req)
company = company_created_res.company
print(company.id, company.name, company.groups[0].id)
```
</TabItem>

<TabItem value="csharp" label="C#">

```c#
var companyCreatedRes = await platformClient.Companies.CreateAsync(new CompanyRequestBody() {
    Name = "Platypus Properties",
    Groups = new List<GroupRef>(){
      new(){ Id = groupId }
    }
  });

if(companyCreatedRes.Company != null) {
    var company = companyCreatedRes.Company;
    logger.LogInformation('{CompanyId} {CompanyName} {GroupId}', company.Id, company.Name, company.Groups[0].Id);
}
```
</TabItem>

<TabItem value="go" label="Go">

```go
ctx := context.Background()
companyCreatedRes, err := platformClient.Companies.Create(ctx, shared.CompanyRequestBody{
    Name: "Platypus Properties",
    Groups: []shared.GroupRef{
      shared.GroupRef{ ID: groupID }
    }
})

if err != nil {
    log.Fatal(err)
}

if companyCreatedRes.Company != nil {
  company := companyCreatedRes.Company
  fmt.Println("%s %s %s", company.Id, company.Name, company.Groups[0].ID)
}
```
</TabItem>

</Tabs>

> If you need to add a company to a group use the [Add company to group](/platform-api#/operations/add-company-to-group) endpoint.

Reviewer: ```suggestion
If you need to add an existing company to a group, use the [Add company to group](/platform-api#/operations/add-company-to-group) endpoint.
```

<Tabs>

<TabItem value="nodejs" label="TypeScript">

```javascript
Add company to group code snippet
```
</TabItem>

<TabItem value="python" label="Python">

```python
Add company to group code snippet
```
</TabItem>

<TabItem value="csharp" label="C#">

```c#
Add company to group code snippet
```
</TabItem>

<TabItem value="go" label="Go">

```go
Add company to group code snippet
```
</TabItem>

</Tabs>

> To remove a company from a group use the [Remove company from group](/platform-api#/operations/remove-company-from-group) endpoint.

Reviewer: ```suggestion
To remove a company from a group, use the [Remove company from group](/platform-api#/operations/remove-company-from-group) endpoint.
```

<Tabs>

<TabItem value="nodejs" label="TypeScript">

```javascript
const companyDeleteResponse = await platformClient.companies.delete({
    companyId: companyCreatedRes.company.id,
  }); 
```
</TabItem>

<TabItem value="python" label="Python">

```python
company_delete_response = platform_client.companies.delete(
  operations.DeleteCompanyRequest(
    company_id=company_created_res.company.id,
  )
)
```
</TabItem>

<TabItem value="csharp" label="C#">

```c#
var companyDeleteResponse = await platformClient.Companies.DeleteAsync(new(){
    CompanyId = companyCreatedRes.Company.Id,
});
```
</TabItem>

<TabItem value="go" label="Go">

```go
ctx := context.Background()
companyDeleteResponse, err := platformClient.Companies.Delete(ctx, operations.DeleteCompanyRequest{
    CompanyID: companyCreatedRes.Company.ID,
})
```
</TabItem>

</Tabs>

## Assign companies to a group

You can group several companies to simplify the management of these companies and their use cases.
If you are grouping companies for the first time, you need to create a group first using the [Create group](/platform-api#/operations/create-group) endpoint.

> You will only have to do this operation once per group so make sure to persist the `groupId` for reuse within your application.

Reviewer: ```suggestion
You only need to do this operation once per group, so retain the `groupId` to reuse it within your application.
```

> If you have an existing set of groups the [List groups](/platform-api#/operations/list-groups) endpoint.

Reviewer: ```suggestion
If you have an existing set of groups, use the [List groups](/platform-api#/operations/list-groups) endpoint to display them.
```
